nncf/common/batchnorm_adaptation.py

```diff
@@ -0,0 +1,107 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Optional
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+from nncf.api.compression import ModelType
+from nncf.common.initialization import NNCFDataLoader
+import nncf.common.factory as factory
+
+
+class BatchnormAdaptationAlgorithmImpl(ABC):
+    """
+    This is the class from which all framework-specific implementations of
+    the batch-norm statistics adaptation algorithm inherit.
+    """
+
+    def __init__(self,
+                 data_loader: NNCFDataLoader,
+                 num_bn_adaptation_steps: int,
+                 num_bn_forget_steps: int,
+                 device: Optional[str] = None):
+        """
+        Initializes the batch-norm statistics adaptation algorithm implementation.
+
+        :param data_loader: NNCF data loader.
+        :param num_bn_adaptation_steps: Number of batches from the training dataset to pass
+            through the model at initialization in order to update batch-norm statistics of
+            the original model.
+        :param num_bn_forget_steps: Number of batches from the training dataset to pass
+            through the model at initialization in order to erase batch-norm statistics of
+            the original model.
+        :param device: Device to perform initialization. If `device` is `None` then the device
+            of the model parameters will be used.
+        """
+        self._data_loader = data_loader
+        self._num_bn_adaptation_steps = num_bn_adaptation_steps
+        self._num_bn_forget_steps = num_bn_forget_steps
+        self._device = device
```

Comment:
Interfaces may not have an `__init__` method or any specified structure other than the method structure.